### ISSUE TITLE
get rid of ad-hoc mkdir usage

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borgstore#47]
We have ad-hoc `mkdir` at quite some places that makes sure base and nesting directories are either present or created.

With a local fs, this is no big thing, but with a slow remote sftp repo, it means quite some useless (because the directory already exists) mkdir calls being repeated over and over again.

So, guess it is better to create all needed base and nesting directories once when the store is created.

Example (locks):

```
[chan 0] stat(b'/Volumes/USB_APFS/borg2-test-fc/locks/cc22550fcd2df7fc')
[chan 0] mkdir(b'/Volumes/USB_APFS/borg2-test-fc/.', 511)
[chan 0] mkdir(b'/Volumes/USB_APFS/borg2-test-fc/locks', 511)
[chan 0] open(b'/Volumes/USB_APFS/borg2-test-fc/locks/xnbnxkfe.tmp', 'w')
[chan 0] open(b'/Volumes/USB_APFS/borg2-test-fc/locks/xnbnxkfe.tmp', 'w') -> 00000000
[chan 0] close(00000000)
[chan 0] posix_rename(b'/Volumes/USB_APFS/borg2-test-fc/locks/xnbnxkfe.tmp', b'/Volumes/USB_APFS/borg2-test-fc/locks/cc22550fcd2df7fc')
```

**Comment by ThomasWaldmann** [comment]
See some remarks about how some cloud storages deal with (empty) directories: #46

**Comment by ThomasWaldmann** [comment]
Issues encountered:
- `Store` does not have `mkdir`, this is only available in the backends
- `nesting.nest` can deal with arbitrary names, not just all-hex hash-like names. if we would keep it that way, we would not know what directories to generate in advance.
- levels config can be incomplete currently at level with default to 0 - if we want to create every dir in advance, levels has to be complete and have all namespaces and related levels.

**Comment by ThomasWaldmann** [comment]
Fixed by #67.

Code assumes that nesting with level > 0 uses all-hex hash-like names.

**Comment by ThomasWaldmann** [comment]
Looks now like this:
```
[chan 0] stat(b'/tmp/b2-repo/data/58/45/584528941ebf8fae1d203b5bb0aa3713dadf7e8bf6995d35a8820d7f0a6edc5e')
[chan 0] open(b'/tmp/b2-repo/data/58/45/azytxkwy.tmp', 'w')
[chan 0] open(b'/tmp/b2-repo/data/58/45/azytxkwy.tmp', 'w') -> 00000000
[chan 0] close(00000000)
[chan 0] posix_rename(b'/tmp/b2-repo/data/58/45/azytxkwy.tmp', b'/tmp/b2-repo/data/58/45/584528941ebf8fae1d203b5bb0aa3713dadf7e8bf6995d35a8820d7f0a6edc5e')
[
```
